### DISCUSSION TITLE
v3.10.0

### DISCUSSION THREAD

**Discussion by oxy** [coder/code-server#3339] [General]
v3.10.0

VS Code v1.56.0

Upgrading is as easy as installing the new version over the old one. code-server
maintains all user data in `~/.local/share/code-server` so that it is preserved in between
installations.

## New Features

⭐ VSCode 1.56, terminals now persist across refreshes (#2276), add several security scanners (#3176, #3177), and add arm64 support for `--link` (#2822)

- feat: minor connections refactor #3178 @code-asher
- feat(security): add code-scanning with CodeQL #3229 @jsjoeio
- feat(ci): add trivy job for security #3261 @jsjoeio
- feat(vscode): update to version 1.56.0 #3269 @oxy
- feat(ci/build): support arm64 for cloud-agent #3294 @oxy
- feat: use ptyHostService #3308 @code-asher

## Bug Fixes

⭐ code-server should now load more predictably, log out correctly, and have improved ratelimiting!

- fix(login): rate limiter shouldn't count successful logins #3141 @jsjoeio
- fix: refactor logout #3277 @code-asher
- fix: add flag for toggling permessage-deflate #3286 @code-asher
- fix: make sure directories exist #3309 @code-asher

## Documentation

⭐ A new changelog, more examples, improved debugging advice, and more detailed maintainership docs.

- docs(FAQ): add mention of sysbox #3087 @bpmct
- docs: add security policy #3148 @jsjoeio
- docs(guide.md): add `caddy` example for serving from sub-path #3217 @catthehacker
- docs: revamp debugging section #3224 @code-asher
- docs(readme): refactor to use codecov shield #3227 @jsjoeio
- docs(maintaining): use milestones over boards #3228 @jsjoeio
- docs(faq): add entry for accessing OSX folders #3247 @bpmct
- docs(termux): add workaround for Android backspace issue #3251 @jsjoeio
- docs(maintaining): add triage to workflow #3284 @jsjoeio
- docs(security): add section for tools #3287 @jsjoeio
- docs(maintaining): add versioning #3288 @jsjoeio
- docs: add changelog #3337 @jsjoeio

## Development

⭐ VSCode now builds in parallel (#3250), code-server now has more comprehensive testing (#2712) and generates coverage reports (#3151), and lots of CI housekeeping.

- fix(update-vscode): add check/docs for git-subtree #3129 @oxy
- refactor(testing): migrate to playwright-test from jest-playwright #3133 @jsjoeio
- refactor(ci): remove unmaintained CI images and update release workflow #3147 @oxy
- chore(ci): migrate from hub to gh #3168 @oxy
- feat(testing): add e2e tests for code-server and terminal #3169 @jsjoeio
- chore(ranger): fix syntax for extension-request #3172 @oxy
- chore(lib/vscode): update netmask #3187 @oxy
- feat(testing): add codecov to generate test coverage reports #3194 @jsjoeio
- feat(testing): add tests for registerServiceWorker #3200 @jsjoeio
- chore(deps): update dependencies with CVEs #3223 @oxy
- refactor(testing): fix flaky terminal test #3230 @jsjoeio
- chore: ignore 15.x @types/node updates #3244 @jsjoeio
- chore(build): compile vscode+extensions in parallel #3250 @oxy
- fix(deps): remove eslint-plugin-jest-playwright #3260 @jsjoeio
- fix(testing): reduce flakiness of terminal.test.ts and use 1 worker for e2e tests #3263 @jsjoeio
- feat(testing): add isConnected check #3271 @jsjoeio
- feat(testing): add test for src/node/constants.ts #3290 @jsjoeio
- feat: test static route #3297 @code-asher
- refactor(ci): split audit from prebuild #3298 @oxy
- chore(lib/vscode): cleanup/update build deps #3314 @oxy
- fix(build): download correct cloud-agent for arch #3331 @oxy
- fix: xmldom and underscore #3332 @oxy

Cheers! 🍻

<hr /><em>This discussion was created from the release <a href='https://github.com/cdr/code-server/releases/tag/v3.10.0'>v3.10.0</a>.</em>

**Comment by ghost** [comment]
`⭐ VSCode 1.56, terminals now persist across refreshes` 🙏 🙇 🥇

**Reply by hgw77** [reply]
`terminals now persist across refreshes` thank you that is awesome 🥇
